rust/yarp-sys/tests/serialize_tests.rs

```diff
@@ -0,0 +1,100 @@
+use std::{ffi::CString, mem::MaybeUninit, path::Path};
+
+use yarp_sys::{
+    yp_buffer_free, yp_buffer_init, yp_buffer_t, yp_node_destroy, yp_parse, yp_parse_serialize,
+    yp_parser_free, yp_parser_init, yp_parser_t, yp_serialize,
+};
+
+fn ruby_file_contents() -> (CString, usize) {
+    let rust_path = Path::new(env!("CARGO_MANIFEST_DIR"));
+    let ruby_file_path = rust_path.join("../../lib/yarp.rb").canonicalize().unwrap();
+    let file_contents = std::fs::read_to_string(ruby_file_path).unwrap();
+    let len = file_contents.len();
+
+    (CString::new(file_contents).unwrap(), len)
+}
+
+#[test]
+fn serialize_test() {
+    let (ruby_file_contents, len) = ruby_file_contents();
+    let source = ruby_file_contents.as_ptr();
+    let mut parser = MaybeUninit::<yp_parser_t>::uninit();
+    let mut buffer = MaybeUninit::<yp_buffer_t>::uninit();
+
+    unsafe {
+        yp_parser_init(parser.as_mut_ptr(), source, len, std::ptr::null());
+        let parser = parser.assume_init_mut();
+        let node = yp_parse(parser);
+
+        if !yp_buffer_init(buffer.as_mut_ptr()) {
+            panic!("Failed to init buffer");
+        }
+
+        let buffer = buffer.assume_init_mut();
+        yp_serialize(parser, node, buffer);
+
+        let serialized = std::slice::from_raw_parts(buffer.value as *mut u8, buffer.length);
+
+        assert_eq!(&serialized[0..4], b"YARP");
+        assert_eq!(serialized[4..5][0], 0); // YP_VERSION_MAJOR
+        assert_eq!(serialized[5..6][0], 4); // YP_VERSION_MINOR
+        assert_eq!(serialized[6..7][0], 0); // YP_VERSION_PATCH
```

Comment:
I'm a bit iffy on these checks since they'll break when the version changes--thoughts?

(also, isn't yarp on 0.5.0, not 0.4.0?)

Reply:
> Might be worth testing it against CARGO_PKG_VERSION or something?

Hmm... I'm not sure I follow. If I understand correctly, the version info there is inserted by `yarp` (C); where would the yarp-sys crate version come into play here?